Sources/PointFree/Admin/Index.swift

```diff
@@ -0,0 +1,138 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+private let adminEmails = [
+  "mbw234@gmail.com",
+  "stephen.celis@gmail.com"
+]
+
+func requireAdmin<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data>
+  ) -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+
+  return { conn in
+    conn
+      |> (adminEmails.contains(get1(conn.data).email.unwrap) ? middleware : redirect(to: .secretHome))
+  }
+}
+
+let adminIndex =
+  requireUser
+    <<< requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(adminIndexView.contramap(lower))
+
+private let adminIndexView = View<(Database.User, Prelude.Unit)> { currentUser, _ in
+  ul([
+    li([
+      a([href(path(to: .admin(.newEpisodeEmail(.show))))], ["Send new episode email"])
+      ])
+    ])
+}
+
+let showNewEpisodeEmailMiddleware =
+  requireUser
+    <<< requireAdmin
+    <| writeStatus(.ok)
+    >-> respond(showNewEpisodeView.contramap(lower))
+
+private let showNewEpisodeView = View<(Database.User, Prelude.Unit)> { currentUser, _ in
+  ul(
+    episodes
+      .sorted(by: ^\.sequence)
+      .map(li <<< newEpisodeEmailRowView.view)
+    )
+}
+
+private let newEpisodeEmailRowView = View<Episode> { ep in
+  p([
+    .text(encode(ep.title)),
+    form([action(path(to: .admin(.newEpisodeEmail(.send(ep.id))))), method(.post)], [
+      input([type(.submit), value("Send email!")])
+      ])
+    ])
+}
+
+let sendNewEpisodeEmailMiddleware: Middleware<StatusLineOpen, ResponseEnded, T2<Episode.Id, Prelude.Unit>, Data> =
+  requireEpisode(notFoundMiddleware: redirect(to: .admin(.newEpisodeEmail(.show))))
+    <<< requireUser
+    <<< requireAdmin
+    <| { conn in pure(conn.map(const((conn.data.second.first.first)))) }
```

Comment:
Does `second.first.first` mean we're nesting the tuples strangely? Don't we prefer the `getn` accessors?

Reply:
yeah this is cause `requireUser` uses `Tuple` and everything else is using `T`. once we change `requireUser` to `T` things will be nicer